### PR TITLE
feat: IDP-1429 initialized gitHub action to check diff on swagger appIO full

### PR DESCRIPTION
**Pull request by Giuseppe-LaManna**
Added gitHub action that runs openApi-merge-cli scripts and checks if diffs exist with appIO full

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

#### Has This Been Tested?

- [ ] Yes
- [ ] No

### Other information
Target: 
```

```

#### How Has This Been Tested?

- [ ] Unit Test Suite
- [ ] Integration Test Suite
- [ ] Api Tests
- [ ] Load Tests

[Link to test results](https://pagopa.atlassian.net/browse/RTD-XXXX)

---

### CODE REVIEW

File: .github/workflows/idpay-openAPI-AppIO-diff.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: 'Check diff on idpay AppIO full swagger'
+
+on:
+  push:
```
Comment (and-mora) [inline .github/workflows/idpay-openAPI-AppIO-diff.yml:4]: does this action triggers only after push (after PR merge), correct? Why not on PR opening?

File: src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl [hunk 2]
```diff
@@ -9,7 +9,7 @@ paths:
   /transaction/{transactionId}:
     get:
       tags:
-        - payment
+        - mergeIgnore
```
Comment (and-mora) [inline src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl:12]: is this API not needed by appio?

**Review by and-mora — CHANGES_REQUESTED** [review]

File: src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl (hunk 2 above)
Comment (Giuseppe-LaManna) [reply src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl:12]: no this API is not exposed to APPIO currently

File: .github/workflows/idpay-openAPI-AppIO-diff.yml (hunk 1 above)
Comment (Giuseppe-LaManna) [reply .github/workflows/idpay-openAPI-AppIO-diff.yml:4]: the check is performed every push, before the merge, only if there are changes in the folders underlying "domains/idpay-app"

File: src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl (hunk 2 above)
Comment (and-mora) [reply src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl:12]: should it stay on a different swagger?

Comment (Giuseppe-LaManna) [reply src/domains/idpay-app/api/idpay_qrcode_payment/io/openapi.qrcode_payment_io.yml.tpl:12]: i know this is correct, i don't know the reasons why it is in this swagger, if it is necessary we will delve into it separately

File: .github/workflows/idpay-openAPI-AppIO-diff.yml (hunk 1 above)
Comment (and-mora) [reply .github/workflows/idpay-openAPI-AppIO-diff.yml:4]: the action need to be executed in the context of a Pull Request, with `on.push` it runs even if no PR is opened or when a PR is merged. Does it sound correct to you?

I suggest using the `on.pull_request` or `on.pull_request_target` strategy in order to make it run only when a PR is opened or updated.

a snippet from the pr_title action: 
```
on:
  pull_request_target:
    types:
      - opened
      - edited
      - synchronize 
 ```

Comment (Giuseppe-LaManna) [reply .github/workflows/idpay-openAPI-AppIO-diff.yml:4]: the reasoning is correct

**Review by and-mora — APPROVED** [review]

**Review by petretiandrea — APPROVED** [review]

**Pull request merged**